ConsoleProject/src/view/Frontend.cs

```diff
@@ -125,13 +126,34 @@ public class Frontend {
             };
             return listBox;
         }
+        
+        private void ParseNewFile(string[] fileList) {
+            if (fileList.Length > 0) { // is this IF required?
```

Comment:
I'm surprised that this doesn't throw a null pointer exception.

Reply:
Not sure what this means or what I should do about it.
This is just to check if there's any file at all being passed to this method.
I could also return if the length is Zero if you prefer.

Reply:
Check ShowForm signature: arguments is an optional parameter. If it is not provided then it'll get initialized as null.

Here you are calling a property on a null object. Which should throw an exception.